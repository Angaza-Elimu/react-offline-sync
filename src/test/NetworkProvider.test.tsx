import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'react-native-testing-library';
import NetworkProvider from '../components/NetworkProvider';

describe('NetworkProvider', () => {
  it('has the correct structure and default props', () => {
    const { output } = shallow(
      <NetworkProvider
        pingInBackground
        httpMethod="HEAD"
        pingInterval={2000}
        pingOnlyIfOffline
        pingServerUrl="whatever"
        pingTimeout={1000}
        shouldPing
      >
        {() => <Text>Test</Text>}
      </NetworkProvider>,
    );
    expect(output).toMatchSnapshot();
  });
});
