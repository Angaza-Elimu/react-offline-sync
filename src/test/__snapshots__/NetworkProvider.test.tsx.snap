// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkProvider has the correct structure and default props 1`] = `
<NetworkConnectivity
  customHeaders={Object {}}
  httpMethod="HEAD"
  onConnectivityChange={[Function]}
  pingInBackground={true}
  pingInterval={2000}
  pingOnlyIfOffline={true}
  pingServerUrl="whatever"
  pingTimeout={1000}
  shouldPing={true}
>
  [Function]
</NetworkConnectivity>
`;
